### ISSUE TITLE
RFTools: Dimensions does not work with VS2

### ISSUE THREAD

**Issue by OwOsucksAwesomeMinecraftbossAlissha6TRM** [ValkyrienSkies/Valkyrien-Skies-2#376]
when making a dimensions with rftools and vs2 being installed, the game crashes

**Comment by copperwarrior** [comment]
Which version of rftools are you using? Can you provide a link to it?

**Comment by OwOsucksAwesomeMinecraftbossAlissha6TRM** [comment]
The rftools dimensions vers is 1.18-8.0.9 
the rftools base vers is 1.18-3.0.11

**Comment by copperwarrior** [comment]
Can you provide a crash report please?

**Comment by OwOsucksAwesomeMinecraftbossAlissha6TRM** [comment]
I can provide one, please hold on a second.

**Comment by OwOsucksAwesomeMinecraftbossAlissha6TRM** [comment]
oh i guess they fixed it, it happened to me a while ago and I thought it would still occur, sorry for wasting your time

**Comment by OwOsucksAwesomeMinecraftbossAlissha6TRM** [comment]
oh thats epix

just joined the world while in a custom dimension, game crashed

although I do have other mods

but that never happened until i installed VS2

now it only came with a log and not crash report
[latest.log](https://github.com/ValkyrienSkies/Valkyrien-Skies-2/files/10473465/latest.log)

**Comment by Rubydesic** [comment]
Can you try getting it to crash again, wait for the game to fully close down and then upload both latest.log and debug.log

**Comment by CraftyZombie** [comment]
Replicated with the following stats:
Forge: 47.3.33
Minecraft: 1.20.1
Mods:

Valkyrien Skies 2.3.0-beta.5
Hyperbox 4.0.2.0
Kotlin for Forge 4.11.0
These are my logs, including latest.log, debug.log, and the crash report:
https://gist.github.com/CraftyZombie/cf8ba4769986aba4c60231b255689d3b

**Comment by TechTastic** [comment]
This seems to be an issue about multiple mods creating dimensions at runtime at a point where VS cannot handle them?

**Comment by CraftyZombie** [comment]
I don’t believe that’s the issue. Any kind of dimension generation during runtime causes this crash. Overload is not necessary.

**Comment by PriestOfFern** [comment]
This isn't caused by dimension generation during runtime. It's caused by dimension/level creation in the wrong tick phase, which is something that should be fixed by mod creators themselves, as doing it is a bad practice in general

**Comment by CraftyZombie** [comment]
yeah, that's better wording, my bad. forgot what runtime was for a second 💀

**Comment by Spectre0987** [comment]
My mod adds them via MinecraftServer#doRunTask. What tick phase should new dimensions be registered in?

**Comment by blockninja124** [comment]
`TickEvent.ServerTickEvent` on `Phase.START` on forge. 

I think the fabric equivalent is `ServerTickEvents.START_SERVER_TICK`

**Comment by blockninja124** [comment]
For the record: RFTools and Supplementaries both no longer crash. There is also the interface `NoVSLevelDuck` which you can implement to prevent VS loading your dimension at all, which is useful for "fake levels" that you want to instantiate briefly and then kill. If you want to implement this interface without depending on VS, you have to mixin into your own mod such as:
```java
@Mixin(FakeServerLevel.class)
public class MixinFakeServerLevel implements NoVSLevelDuck {
}
```
 and conditionally apply that mixin when VS is installed.

**Comment by Dorian4771** [comment]
https://modrinth.com/mod/infinite-dimensions
this crash also happens whenever a dimension is created with this mod, it has been reported by the mod dev themselves yet they're a bit too depressed to fix it, so i think you should just fix the issue on your end because this happens with multiple dimension mods (also fix it for 1.20.1 too)
